### PR TITLE
Feature/add typescript compiler

### LINKED ISSUE #13: Add Typescript compiler into the JS build process.
This should allow for ES2015 syntax if nothing else.

Module loading to be addressed later as part of #11 ?

### PR DESCRIPTION
**Pull request by Chapabu**
Fixes #13

### CODE REVIEW

**Comment by Chapabu** [comment]
This fixes #14

File: src/scripts/buildScripts.js [hunk 1]
```diff
@@ -5,19 +5,32 @@ const path = require('path');
...
-  const jsSrc = path.join(conf.themeConfig.root, conf.themeConfig.js.src, '**', '*.js');
+  const typeScriptConf = {
+    target: 'es5',
+    allowJs: true
+  };
+
+  const jsSrc = path.join(conf.themeConfig.root, conf.themeConfig.js.src, '**', '*.[tj]s');
   const jsDest = path.join(conf.themeConfig.root, conf.themeConfig.js.dest);
 
+  let useTypescript = false;
```
Comment (BrianDGLS) [inline src/scripts/buildScripts.js]: The if statement can be removed here.

```javascript
const useTypescript = Boolean(conf.themeConfig.js.es2015);
```

Comment (Chapabu) [inline src/scripts/buildScripts.js]: Changed, thanks :)

**Comment by BrianDGLS** [comment]
Reviewed 👍

**Pull request merged**